Northstar.Custom/mod/scripts/vscripts/gamemodes/_gamemode_fw.nut

```diff
@@ -1384,19 +1381,23 @@ void function FWAreaThreatLevelThink_Threaded()
 ///// TURRET FUNCTIONS /////
 ////////////////////////////
 
-void function FWTurretHighlight( entity turret )
+void function OnFWTurretSpawned( entity turret )
 {
-    thread FWTurretHighlightThink( turret )
+	turret.EnableTurret() // always enabled
+	SetDefaultMPEnemyHighlight( turret ) // for sonar highlights to work
+	AddEntityCallback_OnDamaged( turret, OnMegaTurretDamaged )
+	thread FWTurretHighlightThink( turret )
 }
 
 // this will clear turret's highlight upon their death, for notifying players to fix them
 void function FWTurretHighlightThink( entity turret )
 {
-    turret.EndSignal( "OnDestroy" )
-    Highlight_SetFriendlyHighlight( turret, "fw_friendly" )
+	turret.EndSignal( "OnDestroy" )
+	WaitFrame()
```

Comment:
Please add a comment as to why is this here :)

Reply:
Once this addressed we should merge the PR ^^